test/tests/p5.FFT.js

```diff
@@ -55,4 +55,61 @@ describe('p5.FFT', function () {
     expect(fft.smoothing).to.equal(0.8);
     expect(fft.smooth()).to.equal(0.8);
   });
+
+  let soundFile, oscillator;
+  it('can set Oscillator as input', function (done) {
+    oscillator = new p5.Oscillator();
+    try {
+      fft.setInput(oscillator);
+    } catch (err) {
+      return expect.fail();
+    }
+    oscillator.start();
+
+    setTimeout(() => {
+      let spectrum = fft.analyze();
+      const someFrequencyIsNotZero = spectrum.some(
+        (frequency) => frequency !== 0
+      );
+      expect(someFrequencyIsNotZero).to.equal(true);
+      oscillator.dispose();
+      done();
+    }, 500);
+  });
+  it('can set soundFile  as input', function (done) {
+    soundFile = p5.prototype.loadSound('./testAudio/drum.mp3', () => {
+      soundFile.disconnect();
+      try {
+        fft.setInput(soundFile);
+      } catch (err) {
+        return done(err);
```

Comment:
it catches an error , right now i am unsure of why we cant set sounfile as input , any suggestions @therewasaguy

Reply:
@davepagurek any thoughts on this?

Reply:
Interesting, I can't actually reproduce this error, it seems to just work. Maybe it was a browser bug that has since been fixed?

Reply:
I will try to reproduce this error, if it works we can merge this one.

Reply:
@davepagurek it was dropping these errors for me : 

![image](https://user-images.githubusercontent.com/64387054/216782276-52d82f87-e527-41b7-a979-f100479b4288.png)

but when I added: `let fft = new p5.FFT();`  it got resolved, anything else seems fine to me.

Reply:
Oh does this happen when running tests headless from the other open PR? I'm running `npm run test` and going to http://localhost:8000/test/?grep=FFT, which is what shows no errors.

Where are you adding the `new p5.FFT()` line? It also has that in a `beforeEach`, but maybe something is up with the asynchronous tests 🤔

Reply:
Ya I am getting that while running the test headless, I added the new line within the function, at the very beginning just like we did for ` oscillator = new p5.Oscillator();`